core/src/blocks/block.rs

```diff
@@ -229,7 +230,15 @@ pub fn replace_variables_in_string(text: &str, field: &str, env: &Env) -> Result
 
     // Run Tera templating engine one_off on the result (before replacing variables but after
     // looking for them).
-    let context = Context::from_value(json!(env.state))?;
+    let mut full_env: HashMap<String, Value> = HashMap::new();
+    for (key, value) in &env.secrets {
```

Comment:
Let's try to do something a bit more rusty

We can iter both and merge iter right? and collect and do that in line 241?